Create a Functions WG

lance opened the pull request:
# Changes

Adds Functions as a distinct working group outside of the Client. It does not remove existing peribolos settings for kn-plugin-func.

## Steps outlined in mechanics/WORKING-GROUP-PROCESSES.md

- [x] **Create a Google Drive Folder**. Create a folder to hold your working group documents within this parent [folder](https://drive.google.com/drive/folders/0AM-QGZJ-HUA8Uk9PVA).  Call your folder "GROUP_NAME".
- [x] **Create a Meeting Notes Document**. Create a blank document in the above folder and call it "GROUP_NAME Group Meeting Notes".
- [x] **Create a Roadmap**. Create a new [Roadmap Project Board](https://github.com/knative/community/blob/main/mechanics/ROADMAPS.md).
- [x] **Schedule a Recurring Meeting**. Create a recurring meeting (weekly or bi-weekly, 30 or 60 minutes) on the [shared calendar](https://calendar.google.com/calendar/u/0/embed?src=knative.team_9q83bg07qs5b9rrslp5jor4l6s@group.calendar.google.com) and call the meeting "\$GROUP_NAME WG". Attach the meeting notes document to  the calendar event. Generally schedule these meetings between 9:00AM to 2:59PM Pacific Time. Invite `knative-dev@googlegroups.com` Google group to the  meeting, as well as necessary individual participants.
- [x] **Register the Working Group**. Go to  [WORKING-GROUPS.md](../working-groups/WORKING-GROUPS.md) and add your working  group name, the names of the leads, the working group charter, and a link to the meeting you created.
- [x] **Create Github Team using Peribolos**. Go to [peribolos/knative.md](../peribolos/knative.md), add your working group name under teams, and members. Grant access to relevant repositories for the working group.
- [x] **Announce your Working Group**. Send a note to [knative-dev@](mailto:knative-dev@googlegroups.com),  [knative-tech-oversight@](mailto:knative-tech-oversight@googlegroups.com), and  [steering@](mailto:steering@knative.team) to announce your new working group. Include your charter in the email and provide links to the meeting invitation.

lance commented:
Regarding
> Create a Roadmap. Create a new [Roadmap Project Board](https://github.com/knative/community/blob/main/mechanics/ROADMAPS.md).

The knative-sandbox/kn-plugin-func repo already has a project at https://github.com/orgs/knative-sandbox/projects/7. If we move the repository I wonder if this project will move with it.

lance commented:
Something should go into knative-sandbox/knobots as well, right?

lance commented:
> https://docs.google.com/document/d/1rwRt4Hi27Jgoqg8wK6tmtVZJdxOFheQei-_Wx3KPsQc/edit
> 
> Proposed charter

Looks great - thanks for doing that! I've got a few comments in the doc.

lance commented:
@evankanderson once we finalize the charter text, I think we are good to bless the working group and announce it. Will be great to have completed by KnativeCon! /cc @salaboy as we discussed, I'm adding you as a WG lead for Functions.

lance commented:
Peribolos failure

```
failed to configure knative members: all team members/maintainers must also be org members: lkingland, matejvasek, zroubalik"
```

lance commented:
/retest-required

lance commented:
/retest-required

lance commented:
/retest-required
Tried manually creating a "Func Reviewers" group to see if that resolves the issue. https://github.com/orgs/knative/teams/func-reviewers

lance commented:
/retest-required

*as he struggles to learn peribolos*

lance commented:
/retest-required

rhuss commented on peribolos/knative.yaml:
```diff
@@ -624,6 +624,12 @@ orgs:
         description: 'DEPRECATED: Development continues in https://github.com/knative/operator/'
         has_projects: true
         has_wiki: false
+      func:
```
Shouldn't this go into `knative-sandbox.yaml` as the repositories are still ? Or is this just mirroring the sandbox configuration ?

lance replied:
@rhuss it has been discussed, but not as a requirement for being a full WG. Do you think it would be better to keep everything as `knative-sandbox` for now? What I am doing here has been additive, not replacing what already exists, fwiw.

rhuss replied:
no problem and no blocker for me.

n3wscott commented:
I am +1 on forming the Functions WG.

evankanderson commented:
/lgtm

evankanderson reviewed (commented):
(non-blocking)

dprotaso commented:
+1 - added some minor comments to the charter

evankanderson commented:
/approve

lance commented:
@evankanderson I think everything is complete here. Should this unhold?

evankanderson commented:
/hold cancel

salaboy commented:
hmmm 
```
{"component":"peribolos","file":"k8s.io/test-infra/prow/cmd/peribolos/main.go:188","func":"main.main","level":"fatal","msg":"Configuration failed: failed to configure knative teams: failed to update Func Reviewers members: failed to list func-reviewers(Func Reviewers) members: return code not 2XX: 404 Not Found","severity":"fatal","time":"2022-06-02T23:56:17Z"}
```

lance commented:
> hmmm
> 
> ```
> {"component":"peribolos","file":"k8s.io/test-infra/prow/cmd/peribolos/main.go:188","func":"main.main","level":"fatal","msg":"Configuration failed: failed to configure knative teams: failed to update Func Reviewers members: failed to list func-reviewers(Func Reviewers) members: return code not 2XX: 404 Not Found","severity":"fatal","time":"2022-06-02T23:56:17Z"}
> ```

I don't really understand. I thought that the configuration declared in these yaml files would create the group if it didn't exist.

dprotaso commented:
peribolos isn't very intelligent - make the team first. Then add members in a separate PR

lance commented:
I have updated this PR to add the charter as a markdown document in this repo instead of a word doc. I have also removed all peribolos configuration changes. These should not be blocking the PR and can be addressed separately.

@knative/steering-committee @knative/technical-oversight-committee ptal

evankanderson reviewed (commented):
/lgtm
/approve
